### PR TITLE
Enable compilation on windows.

### PR DESCRIPTION
**Pull request by gogo2464**
rebase of https://github.com/freenet/freenet-core/pull/956

### CODE REVIEW

File: crates/fdev/src/build.rs [hunk 1]
```diff
@@ -259,65 +274,66 @@ mod contract {
...
-                        .stdout(Stdio::piped())
-                        .stderr(Stdio::piped())
-                        .spawn()
-                        .map_err(|e| {
-                            eprintln!("Error while executing command tsc: {e}");
-                            Error::CommandFailed("tsc")
-                        })?;
-                    pipe_std_streams(child)?;
-                    println!("Compiled input using tsc");
+                    let webpack = web_config
```
Comment (iduartgomez) [inline crates/fdev/src/build.rs:292]: this is redundant if we are checking for webpack already above (line 277);
the else condition of 298 won't event be ever be executed here, so it needs to be moved to an else branch for line 277

**Review by iduartgomez — CHANGES_REQUESTED** [review]
There are some errors

**Comment by gogo2464** [comment]
@iduartgomez I now have this issue from another part of code:

```

error[E0614]: type `freenet_stdlib::prelude::ContractInstanceId` cannot be dereferenced
   --> crates\core\src\contract\executor\runtime.rs:867:22
    |
867 |                 key: *original_key.id(),
    |                      ^^^^^^^^^^^^^^^^^^
```


is it related to my own codebase or is it due to the original freenet codebase?

**Comment by gogo2464** [comment]
see https://github.com/freenet/freenet-core/actions/runs/7617268691/job/20745918244?pr=965#step:7:928

**Comment by iduartgomez** [comment]
> @iduartgomez I now have this issue from another part of code:
> 
> ```
> 
> error[E0614]: type `freenet_stdlib::prelude::ContractInstanceId` cannot be dereferenced
>    --> crates\core\src\contract\executor\runtime.rs:867:22
>     |
> 867 |                 key: *original_key.id(),
>     |                      ^^^^^^^^^^^^^^^^^^
> ```
> 
> is it related to my own codebase or is it due to the original freenet codebase?

Is because the stdlib commit change. That doesn't have to change.

**Comment by gogo2464** [comment]
cool

**Comment by gogo2464** [comment]
how to fix the cicd then?

**Comment by netsirius** [comment]
@gogo2464 I have solved the out-of-sync issue with the stdlib commit. The next step is to resolve the Clippy errors.

**Review by iduartgomez — APPROVED** [review]

**Pull request merged**